federation/pkg/federation-controller/service/dns.go

```diff
@@ -342,12 +343,18 @@ func getResolvedEndpoints(endpoints []string) ([]string, error) {
 				return resolvedEndpoints, err
 			}
 			resolvedEndpoints = append(resolvedEndpoints, ipAddrs...)
-
 		} else {
 			resolvedEndpoints = append(resolvedEndpoints, endpoint)
 		}
 	}
-	return resolvedEndpoints, nil
+	deduped := sets.String{}
```

Comment:
Can't you just make resolvedEndpoints above be a sets.String, and do the deduplication in the above loop, rather than introducing a second loop here?